### DISCUSSION TITLE
Adding copy button is it possible.

### DISCUSSION THREAD

**Discussion by Xeno468** [microsoft/sample-app-aoai-chatGPT#1052] [Q&A]
Hi, 
I want to know if an option is available in order to add a copy button to the ai response. My goal is to have an easier way to copy the response instead of select and copy.

Thanks,

**Comment by FlorianReinprecht** [comment]
I would also be very interested in this - seems to also have been added as an issue as well https://github.com/microsoft/sample-app-aoai-chatGPT/issues/166. 

Adding top contributors for visibility @sarah-widder, @iseabock, @abhahn - maybe you can help us. Thanks!

**Comment by sydneybeckcook** [comment]
I implemented this in Answer.tsx by adding a copy button to the footerIcons:             

```
<Copy24Filled
              title="Copy Answer"
              aria-label="Copy answer text to clipboard"
              onClick={copyAnswerToClipboard}
              className={styles.copyButton}
            />
```

and wrote copyAnswerToClipboard:   

```
const [isCopyNotificationVisible, setCopyNotificationVisible] = useState(false)
const copyAnswerToClipboard = () => {
    if (parsedAnswer?.markdownFormatText) {
      navigator.clipboard.writeText(parsedAnswer.markdownFormatText);
      setCopyNotificationVisible(true);
      setTimeout(() => setCopyNotificationVisible(false), 3000);
    }
  }
``` 
I also added a notification below the footerIcon stack: 

```
        {isCopyNotificationVisible && (
          <MessageBar
            messageBarType={MessageBarType.success}
            isMultiline={false}
            onDismiss={() => setCopyNotificationVisible(false)}
            dismissButtonAriaLabel="Close">
            Response copied to clipboard
          </MessageBar>
        )}
```
